src/cli/lint/internalLints/upgrade/patches/8.2.1/upgradeESLint.ts

```diff
@@ -0,0 +1,126 @@
+// eslint-disable-next-line no-restricted-imports -- fs-extra is mocked
+import * as fsp from 'fs/promises';
+import path from 'path';
+import { inspect } from 'util';
+
+import { promises as fsExtra } from 'fs-extra';
+
+import type { PatchFunction, PatchReturnType } from '../..';
+import { createExec } from '../../../../../../utils/exec';
+import { log } from '../../../../../../utils/logging';
+import { createDestinationFileReader } from '../../../../../configure/analysis/project';
+import { mergeWithConfigFile } from '../../../../../configure/processing/configFile';
+import { formatPrettier } from '../../../../../configure/processing/prettier';
+
+const upgradeESLint: PatchFunction = async ({
+  mode,
+  dir: cwd = process.cwd(),
+}): Promise<PatchReturnType> => {
+  const readFile = createDestinationFileReader(cwd);
+  const [originalIgnoreContents, eslintConfig] = await Promise.all([
+    readFile('.eslintignore'),
+    readFile('.eslintrc.js'),
```

Comment:
suggestion: Is it worth declaring some constants up the top. I notice a few repeats here and there